src/test/java/core/basesyntax/RegistrationValidationTest.java

```diff
@@ -0,0 +1,148 @@
+    void getUser_AfterAddFromStorageClass_Ok() {
+        User user = new User();
+        user.setAge(19);
+        user.setLogin("test007");
+        user.setPassword("qwerty1234");
+        Storage.people.add(user);
+        User getUser = storageDao.get(user.getLogin());
+        assertEquals(user, getUser);
+    }
+    //endregion
+
+    //region assertThrows Tests
+    @Test
+    void register_invalidUser_notOk() {
+        User invalidAgeUser = new User();
+        invalidAgeUser.setAge(17);
+        invalidAgeUser.setLogin("validName");
+        invalidAgeUser.setPassword("validPassword");
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(invalidAgeUser));
+
+        User invalidLoginUser = new User();
+        invalidLoginUser.setAge(18);
+        invalidLoginUser.setLogin("");
+        invalidLoginUser.setPassword("validPassword");
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(invalidLoginUser));
+
+        User invalidPasswordUser = new User();
+        invalidPasswordUser.setAge(19);
+        invalidPasswordUser.setLogin("validName");
+        invalidPasswordUser.setPassword("abc");
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(invalidPasswordUser));
+    }
+
+    @Test
+    void register_existingUser_notOk() {
+        User user = new User();
+        user.setLogin("userTest");
+        Storage.people.add(user);
+        User userLoginCopy = new User();
+        userLoginCopy.setAge(Integer.MAX_VALUE);
+        userLoginCopy.setLogin(user.getLogin());
+        userLoginCopy.setPassword("validPassword");
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(userLoginCopy));
+    }
+
+    @Test
+    void register_ageNotEligible_notOk() {
+        User user = new User();
+        user.setAge(0);
+        user.setLogin("userValid");
+        user.setPassword("password");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_loginShorterThan6_notOk() {
+        User user = new User();
+        user.setAge(55);
+        user.setLogin("user1");
+        user.setPassword("password");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_passwordShorterThan6_notOk() {
+        User user = new User();
+        user.setAge(50);
+        user.setLogin("userValid");
+        user.setPassword("abc");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User user = new User();
+        user.setAge(null);
+        user.setLogin("userValid");
+        user.setPassword("abc123");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));

```

Comment:
The test 'register_nullAge_notOk' is not correct because the 'age' field is of primitive type 'int' and cannot be null. If you want to test an invalid age, consider using a negative number or zero.